File: update-core/src/main/java/dev/bee/kanjianki/updatecore/AutoUpdateRunPolicy.java
package dev.bee.kanjianki.updatecore;

public final class AutoUpdateRunPolicy {
    private AutoUpdateRunPolicy() {
    }

    public static boolean shouldRun(boolean enabled, boolean hasPendingUpdate) {
        return enabled && !hasPendingUpdate;
    }

    public static WorkerOutcome workerOutcome(boolean retryable) {
        return retryable ? WorkerOutcome.RETRY : WorkerOutcome.SUCCESS;
    }

    public enum WorkerOutcome {
        SUCCESS,
        RETRY
    }
}

File: update-core/src/main/kotlin/dev/bee/kanjianki/updatecore/AutoUpdateRunPolicy.kt
package dev.bee.kanjianki.updatecore

object AutoUpdateRunPolicy {
    @JvmStatic
    fun shouldRun(enabled: Boolean, hasPendingUpdate: Boolean): Boolean {
        return enabled && !hasPendingUpdate
    }

    @JvmStatic
    fun workerOutcome(retryable: Boolean): WorkerOutcome {
        return if (retryable) WorkerOutcome.RETRY else WorkerOutcome.SUCCESS
    }

    enum class WorkerOutcome {
        SUCCESS,
        RETRY,
    }
}
